### PR TITLE
Feature/fe 43 view UI implement links page

### PR DESCRIPTION
**Pull request by emmanueletukudo**
### Scope & Purpose
**View UI: Implement "Links" page:**

Currently, there is a nested recursive UI on the links page of the view web UI when a user creates a new link and subsequently adds new fields to both the root and sub-levels of the link. Also, all the links and fields are displayed in a single web page /component, making it difficult for the user to navigate seamlessly. 

This PR simplifies the workflow on the web view UI (Links page), fixes the recursive UI on the link page and allow for users to view a single link or field with their properties at a time. 

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [x] :hankey: Bugfix
- [x] :pizza: New feature
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts [hunk 1]
```diff
@@ -23,8 +23,13 @@ export type DispatchArgs<FormState> = {
   formState?: FormState;
 };
 
+export type RenderView = "LinkList" | "AddNew" | "ViewParent" | "ViewChild" | "ViewField";
```
Comment (adityamukho) [inline js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts:26]: This is views-specific, and should be part of `views/constants.ts`.

File: js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts [hunk 2]
```diff
@@ -23,8 +23,13 @@ export type DispatchArgs<FormState> = {
...
+
 export type FormProps<FormState> = {
   formState: FormState;
   dispatch: Dispatch<DispatchArgs<FormState>>;
   disabled?: boolean;
+  view?: string;
+  show?: RenderView;
```
Comment (adityamukho) [inline js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts:33]: This should be an extension to base form state that is views-specific. The common `utils/constants` should not contain anything that only applies to views.

**Review by adityamukho — CHANGES_REQUESTED** [review]
There are too many formatting changes that do not alter the logic, but make it difficult to review the code. Suggest that those be reverted for this PR.

Re-formats are best grouped together in a separate, dedicated PR.

File: js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts (hunk 1 above)
Comment (emmanueletukudo) [reply js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts:26]: Done.

File: js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts (hunk 2 above)
Comment (emmanueletukudo) [reply js/apps/system/_admin/aardvark/APP/react/src/utils/constants.ts]: Done.

**Review by adityamukho — APPROVED** [review]
LGTM!

**Comment by adityamukho** [comment]
@KVS85 Please merge.

**Review by ajurdak — APPROVED** [review]
LGTM.

**Pull request merged**